languages.yaml

```
@@ -2307,6 +2307,33 @@ Raku:
       - "#FFFFFF"
       - "#00FF39"
     chip: "#0000FB"
+Razor:
+  type: programming
```

```suggestion
  type: markup
```

according to linguist: https://github.com/github-linguist/linguist/blob/1647b5be1658fb15a5e09bf89067f79b392301e2/lib/linguist/languages.yml#L2886

---

By that logic we should probably change Vue's type to "markup". It's "markup" in the permalink provided, and Vue and Razor seem to have a vaguely similar at a high level (a combo of HTML and a programming language).

---

> By that logic we should probably change Vue's type to "markup". It's "markup" in the permalink provided, and Vue and Razor seem to have a vaguely similar at a high level (a combo of HTML and a programming language).

Thats correct.` .razor` files are indeed a combo of HTML and a programming language. I wasn't sure if it should of went in markup, or programming since technically you can run C# code within the markup file with the `@code` block. https://learn.microsoft.com/en-us/aspnet/core/mvc/views/razor has great documentation on what you can do within an `.razor` file

---

In this situation where it can be either markup or programming, because it's a mix of both, I think you should look at what's most important.

For example, with Vue, a `.vue` file is perfectly valid and normal with only HTML and no JS. But a `.vue` file with only JS and no HTML is uncommon.

---

Also, see https://learn.microsoft.com/en-us/aspnet/core/blazor/components/. It goes into more detail about `@code` blocks

---

You're right, I'll fix it